frontend/src/contexts/auth.context.tsx

```diff
@@ -15,6 +15,8 @@ interface ContextProps {
   setAuthenticated: Dispatch<SetStateAction<boolean>>
   email: string
   setEmail: Dispatch<SetStateAction<string>>
+  announcementVersion: number | null | undefined
```

Comment:
Should `announcementVersion` should be added to the `AuthContextProvider`? On one hand `announcementVersion` does need to be visible to all components, since the modal should pop up regardless of which page you decided to navigate to. On the other hand, I don't know if it's considered relevant to 'auth'.